modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationTasks.java

```diff
@@ -57,128 +71,74 @@
-    FunctionConfig<?> functionConfig = functionGroup
-        .map(group -> functionConfig(group, target, effectiveColumn))
-        .orElse(FunctionConfig.missing());
-
-    Map<String, Object> functionArguments = functionGroup
-        .map(ConfiguredFunctionGroup::getArguments)
-        .orElse(ImmutableMap.of());
-
-    return CalculationTask.of(
-        target,
-        measure,
-        rowIndex,
-        columnIndex,
-        functionConfig.createFunction(functionArguments),
-        marketDataMappings,
-        reportingCurrency);
-  }
-
-  /**
-   * Returns configuration for calculating a value.
-   *
-   * @param configuredGroup  the function group providing the function to calculate the value
-   * @param target  the target of the calculation
-   * @param column  the column containing the value. This defines the measure that is calculated
-   * @return configuration for calculating the value
-   */
-  private static <T extends CalculationTarget> FunctionConfig<T> functionConfig(
-      ConfiguredFunctionGroup configuredGroup,
-      CalculationTarget target,
-      Column column) {
-
-    @SuppressWarnings("unchecked")
-    FunctionGroup<T> functionGroup = (FunctionGroup<T>) configuredGroup.getFunctionGroup();
-    Measure measure = column.getMeasure();
-    return functionGroup.functionConfig(target, measure).orElse(FunctionConfig.missing());
+    return new CalculationTasks(taskBuilder.build(), columns);
   }
 
   /**
    * Obtains an instance from a set of tasks and columns.
    * 
-   * @param calculationTasks  the tasks that perform the calculations
+   * @param tasks  the tasks that perform the calculations
    * @param columns  the columns that define the calculations
    * @return the calculation tasks
    */
-  public static CalculationTasks of(List<CalculationTask> calculationTasks, List<Column> columns) {
-    return new CalculationTasks(calculationTasks, columns);
+  public static CalculationTasks of(List<CalculationTask> tasks, List<Column> columns) {
+    return new CalculationTasks(tasks, columns);
   }
 
   //-------------------------------------------------------------------------
   /**
    * Creates an instance.
    * 
-   * @param calculationTasks  the tasks that perform the calculations
+   * @param tasks  the tasks that perform the calculations
    * @param columns  the columns that define the calculations
    */
-  private CalculationTasks(List<CalculationTask> calculationTasks, List<Column> columns) {
+  private CalculationTasks(List<CalculationTask> tasks, List<Column> columns) {
     this.columns = ImmutableList.copyOf(columns);
-    this.calculationTasks = ImmutableList.copyOf(calculationTasks);
+    this.tasks = ImmutableList.copyOf(tasks);
 
-    // Validate the number of tasks and number of columns tally
-    int taskCount = calculationTasks.size();
+    // validate the number of tasks and number of columns tally
+    long taskCount = tasks.stream()
```

Comment:
Probably worth renaming `taskCount` to `cellCount` now there isn't a 1-1 relationship between tasks and cells.